src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -0,0 +1,3017 @@
+        const eprosima::fastcdr::fixed_string<128>& string8_value)
+{
+    AnnotationParameterValue annotation_parameter_value;
+    annotation_parameter_value.string8_value(string8_value);
+    return annotation_parameter_value;
+}
+
+const AnnotationParameterValue TypeObjectUtils::build_annotation_parameter_value(
+        const std::wstring& string16_value)
+{
+    AnnotationParameterValue annotation_parameter_value;
+    annotation_parameter_value.string16_value(string16_value);
+    return annotation_parameter_value;
+}
+
+const AppliedAnnotationParameter TypeObjectUtils::build_applied_annotation_parameter(
+        const NameHash& paramname_hash,
+        const AnnotationParameterValue& value)
+{
+    AppliedAnnotationParameter applied_annotation_parameter;
+    applied_annotation_parameter.paramname_hash(paramname_hash);
+    applied_annotation_parameter.value(value);
+    return applied_annotation_parameter;
+}
+
+void TypeObjectUtils::add_applied_annotation_parameter(
+        AppliedAnnotationParameterSeq& param_seq,
+        const AppliedAnnotationParameter& param)
+{
+#if !defined(NDEBUG)
+    for (const AppliedAnnotationParameter& parameter : param_seq)
+    {
+        if (parameter.paramname_hash() == param.paramname_hash())
+        {
+            throw InvalidArgumentError("Annotation parameter already applied");
+        }
+    }
+#endif // !defined(NDEBUG)
+    param_seq.push_back(param);
+}
+
+const AppliedAnnotation TypeObjectUtils::build_applied_annotation(
+        const TypeIdentifier& annotation_typeid,
+        const eprosima::fastcdr::optional<AppliedAnnotationParameterSeq>& param_seq)
+{
+#if !defined(NDEBUG)
+    applied_annotation_type_identifier_consistency(annotation_typeid);
+    if (param_seq.has_value())
+    {
+        applied_annotation_parameter_seq_consistency(param_seq.value());
+    }
+    if (DomainParticipantFactory::get_instance()->type_object_registry().is_builtin_annotation(annotation_typeid))
+    {
+        throw InvalidArgumentError("Found builtin annotation in custom annotation sequence");
+    }
+#endif // !defined(NDEBUG)
+    AppliedAnnotation applied_annotation;
+    applied_annotation.annotation_typeid(annotation_typeid);
+    applied_annotation.param_seq(param_seq);
+    return applied_annotation;
+}
+
+void TypeObjectUtils::add_applied_annotation(
+        AppliedAnnotationSeq& ann_custom_seq,
+        const AppliedAnnotation& ann_custom)
+{
+#if !defined(NDEBUG)
+    applied_annotation_consistency(ann_custom);
+    for (const AppliedAnnotation& annotation : ann_custom_seq)
+    {
+        if (annotation.annotation_typeid() == ann_custom.annotation_typeid())
+        {
+            throw InvalidArgumentError("Annotation already applied");
+        }
+    }
+#endif // !defined(NDEBUG)
+    ann_custom_seq.push_back(ann_custom);
+}
+
+const AppliedVerbatimAnnotation TypeObjectUtils::build_applied_verbatim_annotation(
```

Comment:
IDL 4.2 - 8.3.5.1 verbatim Annotation
string language default "*";
PlacementKind placement default BEFORE_DECLARATION;

Should this default values be considered somehow?